py/mpconfig.h

```diff
@@ -1216,6 +1216,15 @@ typedef double mp_float_t;
 #define MICROPY_PY_BTREE (0)
 #endif
 
+/*****************************************************************************/
+/* Hooks for extra modules (see MODULES.markdown)                            */
+
+// Enable extra modules. This will be set by py/py.mk, don't set it in a
+// config header file.
+#ifndef MICROPY_HAS_EXTRA_MODULES
+#define MICROPY_HAS_EXTRA_MODULES (0)
```

Comment:
Instead of this config option, you might be able to get away with just using:
```c
#ifndef MICROPY_EXTRA_BUILTIN_MODULES
#define MICROPY_EXTRA_BUILTIN_MODULES
#endif
```
like it's done with `MICROPY_PORT_BUILTIN_MODULES`

Reply:
Sadly that isn't directly possible due to a chicken-and-egg problem at the top of objmodule.c so I've left it in place. It would be nice to get rid of it, however.